types/react/index.d.ts

```diff
@@ -306,7 +306,7 @@ declare namespace React {
         // In the future, if we can define its call signature conditionally
         // on the existence of `children` in `P`, then we should remove this.
         readonly props: Readonly<{ children?: ReactNode }> & Readonly<P>;
-        readonly state: null | Readonly<S>;
```

Comment:
this is not a rollback, you need to remove null from the definition as well

otherwise there will be still errors:

https://user-images.githubusercontent.com/1223799/42028798-a9bda0f2-7acd-11e8-8dc5-c740c2b6d14a.png

> this will still give you might be null errors when you're gonna read the state, which is again, idiomatic typescript - properties need to be defined on class if you gonna use them in advance, or to be more precise -> TS is using state definition from base class as it was not overridden in my component